### PR TITLE
chore(core): countDownLatch respects timeout in the presence of spurious wakeups

### PR DESCRIPTION
**Pull request by jerrinot**
I cannot think of a user-friendly title, it's a basic infra.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/mp/SOCountDownLatch.java|1|1|100.00%|

File: core/src/test/java/io/questdb/test/mp/SOCountDownLatchTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import io.questdb.mp.SOCountDownLatch;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.LockSupport;
+
+public class SOCountDownLatchTest {
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/mp/SOCountDownLatchTest.java:35]: nit: would be also nice to add a test for non-timed `await()` to improve code coverage for `SOCountDownLatch`.

**Review by puzpuzpuz — APPROVED** [review]

File: core/src/test/java/io/questdb/test/mp/SOCountDownLatchTest.java (hunk 1 above)
Comment (jerrinot) [reply core/src/test/java/io/questdb/test/mp/SOCountDownLatchTest.java:38]: done

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/mp/SOCountDownLatch.java|1|1|100.00%|

**Comment by bluestreak01** [comment]
hey @jerrinot does this fix user facing issue? if so, what symptoms?

**Comment by jerrinot** [comment]
@bluestreak01: I could not pinpoint it to any visible user-facing issue. 

It's just a bug in our stdlib. I stumbled upon it while looking at some other issue (which turned out to be a false alarm) 

For now, this is the only usage of this method: https://github.com/jerrinot/questdb/blob/aec8b347ba5f5402b6d7470b64d57ec9fe989dd5/core/src/main/java/io/questdb/cairo/TableWriter.java#L7481 
So the worst-case scenario: The thread might wait longer before it resorts to work-stealing.

**Comment by bluestreak01** [comment]
thanks @jerrinot , i think we should make it a "chore" to help us avoid including cryptic items in the release notes

**Pull request merged**